cloud/scope/machine.go

```diff
@@ -338,6 +338,9 @@ func (m *MachineScope) InstanceSpec(log logr.Logger) *compute.Instance {
 				m.ClusterGetter.Name(),
 			),
 		},
+		Params: &compute.InstanceParams{
+			ResourceManagerTags: infrav1.AddResourceManagerTags(context.TODO(), m.GCPMachine.Spec.ResourceManagerTags),
```

Comment:
We should probably capture the additional permissions required in the docs somewhere:

- roles/resourcemanager.tagAdmin
- roles/resourcemanager.tagUser

Reply:
I think, TagUser role should be enough since listing tags and creating tag bindings are the operations implementation is dependent on.